_inc/lib/debugger/class-jetpack-cxn-tests.php

```diff
@@ -126,11 +135,92 @@ public static function increase_timeout() {
 		return 30; // seconds.
 	}
 
+	/**
+	 * The test verifies the blog token exists, and sends an API request to WP.com to make sure it's valid.
+	 *
+	 * @return array
+	 */
+	protected function test__validate_blog_token() {
+		$blog_token = $this->helper_get_blog_token();
+
+		if ( ! $blog_token ) {
+			return self::failing_test(
+				array(
+					'name'              => __FUNCTION__,
+					'short_description' => __( 'Blog token is missing.', 'jetpack' ),
+					'action'            => admin_url( 'admin.php?page=jetpack#/dashboard' ),
+					'action_label'      => __( 'Disconnect your site from WordPress.com, and connect it again.', 'jetpack' ),
+				)
+			);
+		}
+
+		$body = array(
+			'client_id'     => Jetpack_Options::get_option( 'id' ),
+			'client_secret' => $blog_token->secret,
+		);
+
+		$args = array(
+			'method'  => 'POST',
+			'body'    => $body,
+			'headers' => array(
+				'Accept' => 'application/json',
+			),
+		);
+
+		add_filter( 'http_request_timeout', array( $this, 'increase_timeout' ), PHP_INT_MAX - 1 );
```

Comment:
We need to remove the filter after the request a la https://github.com/Automattic/jetpack/blob/master/_inc/lib/debugger/class-jetpack-cxn-tests.php#L378

But, what's the reason for increasing the timeout? WordPress.com should be able to respond quickly without 30 seconds.

The 30 seconds is added for the `test__wpcom_connection_test` test because that test runs a number of tests (the same collection with the jetpack.com debugger) and reports back the status. That suite's number of tests that reach out to the site is the reason for the timeout bump.

As an alternative, could the blog token test be a test from the jetpack.com testing suite instead and, if the blog token is not valid per WP.com, then it is reported on jetpack.com instead?

That would save the extra request -- the idea for the local testing suite is it tests local conditions, with the `test__wpcom_connection_test` used to test the WP.com side of things. Making sure the token is present fits, IMO.